src/linker/Linker/WarningSuppressionWriter.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Mono.Cecil;
+
+namespace Mono.Linker
+{
+	public class WarningSuppressionWriter
+	{
+		private readonly LinkContext _context;
+		private readonly Dictionary<AssemblyNameDefinition, HashSet<(int, IMemberDefinition)>> _warnings;
+
+		public WarningSuppressionWriter (LinkContext context)
+		{
+			_context = context;
+			_warnings = new Dictionary<AssemblyNameDefinition, HashSet<(int, IMemberDefinition)>> ();
+		}
+
+		public void AddWarning ((int, IMemberDefinition) warning)
```

Comment:
Nit: Please make the method accept two parameters instead - using tuple here doesn't seem to make much sense.